### ISSUE TITLE
How to directly use DNA-Rendering data in 3DGS?

### ISSUE THREAD

**Issue by DNA-Rendering** [DNA-Rendering/DNA-Rendering#12]
For users' convenience, we provide data reader functions based on [3DGS](https://github.com/graphdeco-inria/gaussian-splatting) to load data from SMC files directly, rather than to preprocess our data with COLMAP. Please check ["scripts/3DGS"](https://github.com/DNA-Rendering/DNA-Rendering/tree/main/scripts/3DGS) for your reference. 

Let me briefly introduce the changes we made. It loads one pose's multi-view images to train, and save testing views for evaluation. 

**"scripts/3DGS/SMCReaders.py"** is necessary to load SMC files.

In **readDNARenderingInfo()**, it calls **readCamerasDNARendering()** to load necessary data and uses the SMPLx vertices as initial points for 3DGS. Views for testing and training are hard-coded, please modify them for your convenience.
https://github.com/DNA-Rendering/DNA-Rendering/blob/1e41b621f13072400401ca787b7182492f7c3374/scripts/3DGS/dataset_readers.py#L352-L388

In **readCamerasDNARendering()**, it loads SMPLx parameters and the RGB images, masks RGB images without background, converts the OpenCV cameras to the usable version of 3D gaussian splatting. 
https://github.com/DNA-Rendering/DNA-Rendering/blob/1e41b621f13072400401ca787b7182492f7c3374/scripts/3DGS/dataset_readers.py#L265-L350

Finally, we add the necessary entries to integrate into the original 3DGS code base.
https://github.com/DNA-Rendering/DNA-Rendering/blob/1e41b621f13072400401ca787b7182492f7c3374/scripts/3DGS/dataset_readers.py#L390-L394
https://github.com/DNA-Rendering/DNA-Rendering/blob/1e41b621f13072400401ca787b7182492f7c3374/scripts/3DGS/__init__.py#L48-L50

Basically you could replace "dataset_readers.py" and "\_\_init\_\_.py" in [3DGS/scene](https://github.com/graphdeco-inria/gaussian-splatting/tree/main/scene) with our scripts.  

To use 3DGS with our smc file, please put the smc files as following structure:
```
|-- root_folder
    |-- main
         |-- xxxx_xx.smc     ### main smc file 
     |-- annotations
         |-- xxxx_xx_annots.smc    ### annotation smc file
```
**Training command:** `python train.py -s <path to main smc file>`

Hope it helps!

**Comment by p-Cyan** [comment]
Hello Authors!

I was experimenting with the given code. However the outputs were bad. The images do not seem to be 3D consistent with the cameras. This was for 0816_12.smc.

https://github.com/DNA-Rendering/DNA-Rendering/assets/29049626/edc33c68-62d0-4422-87cf-e2af95be9a41

**Comment by DNA-Rendering** [comment]
Hi @p-Cyan ,

Thank you for your interest in DNA-Rendering dataset!

For the issue you mentioned, it is because the original version of gaussian splatting uses the center point as the principal point by default. However, the principal point is defined as other value in our dataset, then the misalignment occurs due to wrong projection matrix. 

In order to get the correct projection matrix, we update the definition of class **Camera** and provided a new function **getProjectionMatrix2()**. Scripts in [utils](https://github.com/DNA-Rendering/DNA-Rendering/tree/main/scripts/3DGS/utils) are based on the modification of code in [3DGS/utils](https://github.com/graphdeco-inria/gaussian-splatting/tree/main/utils) .

https://github.com/DNA-Rendering/DNA-Rendering/blob/114a6d05ace90d32fa0bfdb1b05849ec72bbd458/scripts/3DGS/dataset_readers.py#L34-L46
https://github.com/DNA-Rendering/DNA-Rendering/blob/114a6d05ace90d32fa0bfdb1b05849ec72bbd458/scripts/3DGS/utils/graphics_utils.py#L73-L95

Then, we add the necessary entries to calculate correct projection matrix by using camera intrinsic matrix.
https://github.com/DNA-Rendering/DNA-Rendering/blob/114a6d05ace90d32fa0bfdb1b05849ec72bbd458/scripts/3DGS/dataset_readers.py#L347-L349
https://github.com/DNA-Rendering/DNA-Rendering/blob/114a6d05ace90d32fa0bfdb1b05849ec72bbd458/scripts/3DGS/utils/camera_utils.py#L49-L54
https://github.com/DNA-Rendering/DNA-Rendering/blob/114a6d05ace90d32fa0bfdb1b05849ec72bbd458/scripts/3DGS/cameras.py#L61

Below is our 3DGS results by training with 30 views and 30k iterations, and testing with 18 views. 

https://github.com/DNA-Rendering/DNA-Rendering/assets/136057575/90027f00-8d81-40d9-96fa-b977bf756729

https://github.com/DNA-Rendering/DNA-Rendering/assets/136057575/6831e4fd-944d-4b47-ab35-0e89c11f08da

**Comment by DNA-Rendering** [comment]
We also tested our data on [2DGS](https://github.com/hbb1/2d-gaussian-splatting).  
From the results shown below, 2DGS constructs the background although it has been removed by mask. Therefore, we simply added a bounding box to filter the background region during rendering, some parts could remove successfully, while others not. 

https://github.com/DNA-Rendering/DNA-Rendering/assets/136057575/f466ecb1-6c49-4603-8da1-38e1532da7ba


https://github.com/DNA-Rendering/DNA-Rendering/assets/136057575/455506fa-a676-48df-b4ec-e7e255cb7ee4

**Comment by Shiyu-Xing** [comment]
Hello author. Your testing with 3DGS on DNA dataset is very interesting. Recently, I'm doing the similar experiment with our custom data. We segment out the person in our images and remain the background black. It's fine if you render the model with background black again, but when you switch the background to other color,  I see many black artifacts stick to  the reconstructed person model. Seems like when gaussian training, the black background is also fitted by som small black gaussians, especially near of the edge of segmentation. Do you have any hints on how to counter this effect?  
Many thanks !
